closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -61,3 +82,85 @@ void * find_crypto_symbol(void *handle, const char *symname) {
     return symptr;
 }
 
+/*
+ * Find path to JRE based on .exe's location
+ */
+jboolean
+GetJREPath(char *path, jint pathsize)
+{
+    char javadll[MAX_PATH];
+    struct stat s;
+
+    if (GetApplicationHome(path, pathsize)) {
+        /* Is JRE co-located with the application? */
+        JLI_Snprintf(javadll, sizeof(javadll), "%s\\bin\\" JAVA_DLL, path);
```

Comment:
The return value should be checked to verify that the path is complete.

Reply:
Updated to check return code